### PR TITLE
Configuration as executable code

### PR DESCRIPTION
**Pull request by MatthewHambley**
The ever increasing number of `eval` statements necessary to get useful configuration out of the `.ini` file previously implemented lead me to wonder if we shouldn't just specify our configuration as code.

There is an inherent security issue with this approach in that someone could but an `rm -r *` in the configuration script and stylist would faithfully run it. I think, on balance, this is not something we need to worry about. The people who will be running stylist will largely be the developers of a project so they will only be exposing their own systems. It is once again a reminder to be wary of running random code off the Internet.

### CODE REVIEW

File: unit-tests/engine_test.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 class _StyleHarness(Style):
     def __init__(self) -> None:
-        super(_StyleHarness, self).__init__([])
+        super(_StyleHarness, self).__init__()
```
Comment (Jamulus) [inline unit-tests/engine_test.py:21]: While still valid, `super().__init__()` is preferred in Python 3. This same pattern appears in LanguageHarness in source_test.py

**Review by Jamulus — COMMENTED** [review]
Overall design seems solid, just a few minor points that could be dealt with. I've commented the details. 

Worth mentioning this is my first GitHub code review so if there is any formal guidance on how to code review for this repo I haven't seen it, I've just picked out things that seemed incorrect or confusing.

**Comment by MatthewHambley** [comment]
> Worth mentioning this is my first GitHub code review so if there is any formal guidance on how to code review for this repo I haven't seen it, I've just picked out things that seemed incorrect or confusing.

See https://github.com/MetOffice/stylist/wiki/Guide-for-Reviewers

File: source/stylist/style.py [hunk 2]
```diff
@@ -69,72 +64,3 @@ def _all_subclasses(cls: Any) -> Set[Type]:
                 to_examine.append(child)
                 children.add(child)
     return children
```
[the remark this answers is not in the document]
Comment (MatthewHambley) [reply source/stylist/style.py:66]: Good spot

File: documentation/source/user-manual/preparation.rst [hunk 3]
```diff
@@ -17,38 +17,62 @@ Getting started
...
-``implicit`` statements and procedure arguments without ``intent`` properties.
+The Style object is constructed with all the rules which make up that style.
+To add a second rule just import it and append ``, MissingIntent()`` to the
+argument list::
+
+    from stylist.fortran import MissingImplicit
+    from stylist.style import Style
+
+    simple = Style(MissingImplicit(), MissingIntent())
+
```
[the remark this answers is not in the document]
Comment (MatthewHambley) [reply documentation/source/user-manual/preparation.rst:45]: Added

File: unit-tests/engine_test.py (hunk 1 above)
Comment (MatthewHambley) [reply unit-tests/engine_test.py:21]: Quite correct. Obviously a leftover from when we targeted Python2.

**Review by Jamulus — APPROVED** [review]
Changes seem good, approved.

**Pull request merged**